src/layers/CementLayer.ts

```diff
@@ -48,8 +48,13 @@ export class CementLayer extends WellboreBaseComponentLayer {
     const texture: Texture = this.createTexture();
 
     cementShapes.forEach((cementShape: CementShape) => {
```

Comment:
This is quite ok since there are only two render types.

An alternative way that scales better with more render types is a variant of the strategy pattern:

```
const drawStrategies = {
  [RENDERER_TYPE.CANVAS]: cementShape => {
    this.drawBigTexturedPolygon(cementShape.leftPolygon, texture);
    this.drawBigTexturedPolygon(cementShape.rightPolygon, texture);
  },
  [RENDERER_TYPE.SOMETHINGELSE]: cementShape => {
    this.drawRopeWithMask(cementShape.path, cementShape.leftPolygon, texture);
    this.drawRopeWithMask(cementShape.path, cementShape.rightPolygon, texture);
  }
};

cementShapes.forEach(cementShape => drawStrategies[this.renderType](cementShape));
```

Reply:
I kind of agree that the pattern where there is a test for which type of rendering to use is less than idea but not too much of a problem as it's not that many but essentially it's duplicate code.
An alternative might be a factory pattern with two different classes(pixi, pixy legacy) with implementations of drawRope(), instantiate a member from factory with rendertype as parameter and call drawRope() on implementation.

Reply:
I like the idea of having two renderers with one API and select the correct renderer based on RENDER_TYPE. But I think we can wait with that until we potentially encounter more need for it, and the extra complexity will start paying off.